libraries/ESP8266WiFi/src/WiFiClientSecureBearSSL.cpp

```diff
@@ -437,12 +437,16 @@ int WiFiClientSecure::_run_until(unsigned target, bool blocking) {
     DEBUG_BSSL("_run_until: Not connected\n");
     return -1;
   }
-  for (int no_work = 0; blocking || no_work < 2;) {
-    if (blocking) {
-      // Only for blocking operations can we afford to yield()
-      optimistic_yield(100);
+  
+  unsigned long startMillis = millis();
```

Comment:
This timeout really should use a `polledTimeout`.  See the examples for info.

Reply:
@johnm545 FYI [polledTimeout example](https://github.com/esp8266/Arduino/tree/master/libraries/esp8266/examples/BlinkPolledTimeout)

Reply:
IMO the conventional timeout is just as clear.  Is there another reason to prefer the polledTimeout? Looks like it will result in more variables on the stack, and ram is precious when establishing secure connections.

Reply:
The BSSL stack is already paid for (allocated on first BSSL object creation), so an extra 4 bytes there is cost-free.

We're trying to reduce technical debt in the code, because many folks have trouble implementing timeouts.  So as we update sections where timeouts are implemented (not many in the core, but a few in the examples), we really try to keep a consistent usage.

Reply:
Makes sense - have pushed another commit with polledTimeout using _timeout in constructor.  This works fine against my MCVE and against the BearSSL device test suite.  I will have a crack at setting up a test when time permits.  I think the best way would be an on-device test, with a little python server running on the host that just drops with connection without closing the socket.